lib/ethereum-sync-protocol/src/lib.rs

```diff
@@ -386,441 +386,310 @@ pub fn next_sync_committee_gindex_at_slot<C: ChainSpec>(chain_id: u64, slot: Slo
-//         assert_eq!(
-//             verify_storage_proof(
-//                 VALID_PROOF2.storage_root,
-//                 VALID_PROOF2.storage_proof.key,
-//                 &rlp::encode(&VALID_PROOF2.storage_proof.value),
-//                 VALID_PROOF2.storage_proof.proof.iter()
-//             ),
-//             Ok(())
-//         );
-//     }
-
-//     #[test]
-//     fn is_valid_light_client_header_works() {
-//         UPDATES.iter().for_each(|(_, update)| {
-//             // Both finalized and attested headers should be verifiable
-//             assert_eq!(
-//                 is_valid_light_client_header(&SEPOLIA.fork_parameters, &update.attested_header),
-//                 Ok(()),
-//                 "invalid attested header"
-//             );
-
-//             assert_eq!(
-//                 is_valid_light_client_header(&SEPOLIA.fork_parameters, &update.finalized_header),
-//                 Ok(()),
-//                 "invalid finalized header"
-//             );
-//         });
-//     }
-// }
+#[cfg(test)]
+mod tests {
+    use beacon_api_types::chain_spec::Mainnet;
+    use hex_literal::hex;
+    use lazy_static::lazy_static;
+
+    use super::*;
+
+    pub struct BlsVerifier;
+
+    impl BlsVerify for BlsVerifier {
+        const INVERSE: bool = false;
+
+        fn aggregate_verify_signature<'pk>(
+            &self,
+            _aggregate_public_key: &'pk H384,
+            _public_keys: impl IntoIterator<Item = &'pk H384>,
+            _msg: Vec<u8>,
+            _signature: H768,
+        ) -> Result<(), Error> {
+            Ok(())
+        }
+    }
+
+    const SEPOLIA_CHAIN_ID: u64 = 11155111;
+
+    lazy_static! {
+        pub static ref UPDATE_6553725: ethereum_sync_protocol_types::LightClientUpdate =
+            serde_json::from_str(&include_str!("./test/light_client_update_6553725.json")).unwrap();
+        pub static ref SYNC_COMMITTEE_6553725: SyncCommittee =
+            serde_json::from_str(include_str!("./test/sync_committee_6553725.json")).unwrap();
+    }
+
+    pub const GENESIS_VALIDATORS_ROOT_6553725: H256 = H256::new(hex!(
+        "d8ea171f3c94aea21ebc42a1ed61052acf3f9209c00e4efbaaddac09ed9b8078"
+    ));
+
+    fn update_6553725(
+        update: &ethereum_sync_protocol_types::LightClientUpdate,
+    ) -> Result<(), Error> {
+        validate_light_client_update::<Mainnet, BlsVerifier>(
+            SEPOLIA_CHAIN_ID,
+            update,
+            Some(&SYNC_COMMITTEE_6553725),
+            None,
+            Slot::new(6553726),
+            Slot::new(6553718),
+            GENESIS_VALIDATORS_ROOT_6553725,
+            BlsVerifier,
+        )
+    }
+
+    #[test]
+    fn validate_update_works() {
+        assert_eq!(update_6553725(&UPDATE_6553725), Ok(()));
+    }
+
+    #[test]
+    fn validate_update_fails_when_sync_committee_is_wrong() {
+        let mut update = UPDATE_6553725.clone();
+        update.sync_aggregate.sync_committee_bits = Vec::new();
```

Comment:
this should technically fail, since the vector length is incorrect, so:

- add a test that ensures the sync committee bits vector length is checked correctly
- use https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.clear here instead